packages/opentelemetry-metrics/test/Meter.test.ts

```diff
@@ -683,6 +684,143 @@ describe('Meter', () => {
     });
   });
 
+  describe('#SumObserverMetric', () => {
+    it('should create an Sum observer', () => {
+      const sumObserver = meter.createSumObserver('name') as SumObserverMetric;
+      assert.ok(sumObserver instanceof Metric);
+    });
+
+    it('should return noop observer when name is invalid', () => {
+      const spy = sinon.stub(meter['_logger'], 'warn');
+      const sumObserver = meter.createSumObserver('na me');
+      assert.ok(sumObserver === api.NOOP_SUM_OBSERVER_METRIC);
+      const args = spy.args[0];
+      assert.ok(
+        args[0],
+        'Invalid metric name na me. Defaulting to noop metric implementation.'
+      );
+    });
+
+    it('should create observer with options', () => {
+      const sumObserver = meter.createSumObserver('name', {
+        description: 'desc',
+        unit: '1',
+        disabled: false,
+      }) as SumObserverMetric;
+      assert.ok(sumObserver instanceof Metric);
+    });
+
+    it('should set callback and observe value ', async () => {
+      let counter = 0;
+      function getValue() {
+        counter++;
+        if (counter % 2 === 0) {
+          return -1;
+        }
+        return 3;
+      }
+      const sumObserver = meter.createSumObserver(
+        'name',
+        {
+          description: 'desc',
+        },
+        (observerResult: api.ObserverResult) => {
+          // simulate async
+          return new Promise(resolve => {
+            setTimeout(() => {
+              observerResult.observe(getValue(), { pid: '123', core: '1' });
+              resolve();
+            }, 1);
+          });
+        }
+      ) as SumObserverMetric;
+
+      let metricRecords = await sumObserver.getMetricRecord();
+      assert.strictEqual(metricRecords.length, 1);
+      let point = metricRecords[0].aggregator.toPoint();
+      assert.strictEqual(point.value, 3);
+      assert.strictEqual(
+        hashLabels(metricRecords[0].labels),
+        '|#core:1,pid:123'
+      );
+
+      metricRecords = await sumObserver.getMetricRecord();
+      assert.strictEqual(metricRecords.length, 1);
+      point = metricRecords[0].aggregator.toPoint();
+      assert.strictEqual(point.value, 2);
```

Comment:
AFAIK `SumObserver` value should never fall throughout the lifetime process. Here, it does not look the case. First value was 3 and then dropped to 2.

Reply:
Whoopsies! Should be fixed now.